src/Libraries/Auth/ApiAuth.php

```diff
@@ -89,14 +90,26 @@ public function signin($mailer, $username, $password)
             throw new AuthException(ExceptionMessages::INACTIVE_ACCOUNT);
         }
 
-        if (config()->get('two_step_verification')) {
+        if (filter_var(config()->get('2SV'), FILTER_VALIDATE_BOOLEAN)) {
 
-           $user = $this->towStepVerification($mailer, $user);
-        }
+            $otp_token = $this->generateOtpToken($user[$this->keys['usernameKey']]);
 
-        $tokens = $this->setUpdatedTokens($user);
+            $time = new \DateTime();
 
-        return $tokens;
+            $time->add(new \DateInterval('PT' . config()->get('otp_expiry_time') . 'M'));
+
+            $otp_expiry_time = $time->format('Y-m-d H:i');
+
+            $this->towStepVerification($mailer, $user, $otp_expiry_time, $otp_token);
+
+            return $otp_token;
+
+        } else {
+
+            $tokens = $this->setUpdatedTokens($user);
+
+            return $tokens;
```

Comment:
The expression return $otp_token returns the type string which is incompatible with the documented return type array.